src/screens/crew6/scienceScreen.cpp

```diff
@@ -223,24 +221,18 @@ void ScienceScreen::onDraw(sf::RenderTarget& window)
     GuiOverlay::onDraw(window);
     P<ScanProbe> probe;
 
-    // Handle mouse wheel
-    float mouse_wheel_delta = InputHandler::getMouseWheelDelta();
-    if (mouse_wheel_delta != 0.0 && my_spaceship)
```

Comment:
Removing the conditional means the view distance, zoom slider, and zoom label are recalculated every onDraw tick, even when nothing has changed.

Is there a way for script or tweak menu radar range changes to signal/trigger/callback an update, instead of updating every tick?

Reply:
I think that callbacks arent pratical due to it being on the client and the callbacks being server side, I think it would end up looking like
```
    float view_distance = science_radar->getDistance();
    float mouse_wheel_delta=InputHandler::getMouseWheelDelta();
    if (mouse_wheel_delta!=0)
      view_distance *= (1.0 - (mouse_wheel_delta * 0.1f));
    view_distance = std::min(view_distance,my_spaceship->getLongRangeRadarRange());
    view_distance = std::max(view_distance,my_spaceship->getShortRangeRadarRange());
    if (view_distance!=science_radar->getDistance())
    {
      science_radar->setDistance(view_distance);
      // Keep the zoom slider in sync.
      zoom_slider->setValue(view_distance)->setRange(my_spaceship->getLongRangeRadarRange(),my_spaceship->getShortRangeRadarRange());
      zoom_label->setText(tr("Zoom: {zoom}x").format({{"zoom", string(my_spaceship->getLongRangeRadarRange() / view_distance, 1)}}));
    }
```
I think it is easier to read without the branching logic, and I think the performance impact is likely to be minimal, with the possible exepction of the translation - I dont know how heavy that is.

If people think the above is better I am happy to edit and change and test it though

Reply:
Yeah, I don't see anything egregious here. Of the crew stations that tend to strain lower-end systems, science is the heaviest, so my concern is mostly about adding to that.

How often are people changing radar view distance on ships in the middle of a game?

Reply:
I don't feel strongly on it so I will change it.
In testing it there is also a need to cache the last values for long and short range radar and compare those as well.
I kind of wonder if the logic to check if it needs updating may consume more time than the update itself, it feels branchy but without knowing how expensive the tr call is I have no instinct in the matter (and strings can be expensive).

I have seen small issues with the scanning popup locally - they felt very gui related, I don't know if they are the same ones as low end machines though.

Reply:
(I didn't mean "How often are people changing..." to come off as contrarian, sorry, I genuinely don't know)

Reply:
I don't have any serious concerns with the PR, and if it fixes a problem, it's worth having

Reply:
Overhead of these checks is really low.